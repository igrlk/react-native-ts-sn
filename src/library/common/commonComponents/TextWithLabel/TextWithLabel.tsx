import React from 'react';
import { View, Text } from 'react-native';

import styles from './textWithLabelStyles';

interface TextWithLabelProps {
	text: string;
	label: string;
}

export default function TextWithLabel({ text, label }: TextWithLabelProps) {
	return (
		<View style={styles.container}>
			<Text style={styles.label}>{label}</Text>
			<Text style={styles.text}>{text}</Text>
		</View>
	);
}
