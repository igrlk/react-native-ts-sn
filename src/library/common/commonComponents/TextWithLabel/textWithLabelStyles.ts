import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	container: {
		display: 'flex',
		justifyContent: 'flex-start',
		marginBottom: 18,
		width: 145,
	},
	label: {
		fontFamily: 'Poppins',
		fontSize: 14,
		lineHeight: 21,
		color: '#9E9E9E',
	},
	text: {
		fontFamily: 'Poppins',
		fontSize: 16,
		lineHeight: 24,
		color: '$colorBlack',
	},
});
