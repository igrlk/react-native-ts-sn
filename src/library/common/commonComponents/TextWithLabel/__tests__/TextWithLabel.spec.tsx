import React from 'react';
import renderer from 'react-test-renderer';

import TextWithLabel from '../TextWithLabel';

describe('TextWithLabel', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<TextWithLabel text='text' label='label' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
