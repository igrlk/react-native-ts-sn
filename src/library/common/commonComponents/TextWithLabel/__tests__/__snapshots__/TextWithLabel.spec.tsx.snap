// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithLabel match to snapshot 1`] = `
<View>
  <Text>
    label
  </Text>
  <Text>
    text
  </Text>
</View>
`;
